src/server/server_family.cc

```diff
@@ -1976,6 +2061,39 @@ void ServerFamily::ReplicaOf(CmdArgList args, ConnectionContext* cntx) {
   }
 }
 
+void ServerFamily::ReplicaOf(CmdArgList args, ConnectionContext* cntx) {
+  ReplicaOfInternal(args, cntx, true);
+}
+
+void ServerFamily::Replicate(string host_s, string port_s, bool* success) {
+  io::StringSink sink;
+  ConnectionContext ctxt{&sink, nullptr};
+
+  vector<MutableSlice> vargs{
+      {host_s.data(), host_s.size()},
+      {port_s.data(), port_s.size()},
+  };
+
+  CmdArgList args{vargs.data(), vargs.size()};
+
+  // we don't flush the database as the context is null
```

Comment:
And how do you know that there is nothing to flush? What would happen if I accidently start a df replica instance with the following two flags? 

```
./dragonfly --dbfilename=some_path_to_file  --replicaof some_host_port_here
```

IMO unless I am missing something I don't think it's worth not flushing on this execution path as the overhead will be negligible and things could go wrong when certain flags are improperly messed up

Reply:
The flag begins replication before loading any snapshots, this is done deliberately in order to avoid what you just described :smile: